### PR TITLE
send test correction reason to RS

### LINKED ISSUE #2913: DATA ISSUE: Mark as error
We have received two notes from states/ users regarding the way "Mark as error" is passed down. There is a previous ticket from a user in Illinois, and the RS team received another note from Montana, pasted below. This ticket is to prompt investigation between SR/ Better Data and ReportStream and it can turn into a more specific action after the issue is explored, if there is action required by SimpleReport

MONTANA EMAIL
Hi Maurice,

My name is Danny Power, I’m the new ELR Coordinator for Montana working closely with Sam on managing our database and messages.

I wanted to pick your brain about having Simple Report labs update in NBS if they get edited in the Simple Report platform. Recently we had some facilities enter labs through Simple Report incorrectly, and then use the “Mark as Error” function to note that they were data entry errors. However, the original input still crossed into the state database with no note about the results being wrong.

Is there anyway to have the “error” function in Simple Report trigger a lab update that will cross into the database and mark the previously reported lab as cancelled? I know there are ways to do this with other ELR setups, that way the health departments won’t try to follow up on inaccurate results.

Thanks for your time and let me know if you need any other info!

### PR DESCRIPTION
**Pull request by zdeveloper**
## Related Issue or Background Info
- fixes #2913 

## Changes Proposed

- send ReportStream `Test_correction_reason`
- send corrected test events to report stream (we were not before 🙈)
- added test coverage for putting test events on the storage queue, when submitting test and when correcting a test

### Infrastructure
- [X] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [X] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [X] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [X] Any content changes (including new error messages) have been approved by content team

### Support
- [X] Any changes that might generate new support requests have been flagged to the support team
- [X] Any changes to support infrastructure have been demo'd to support team

### Testing
- [X] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [X] Database changes are submitted as a separate PR
  - [X] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [X] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [X] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [X] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [X] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [X] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [X] Any dependencies introduced have been vetted and discussed

## Cloud
- [X] DevOps team has been notified if PR requires ops support
- [X] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java [hunk 1]
```diff
@@ -334,16 +334,9 @@ public TestEvent createTestEvent(Person p, Facility f, TestResult r, Boolean has
...
-
-    TestEvent te2 =
-        _testEventRepo.save(new TestEvent(te, TestCorrectionStatus.CORRECTED, "Corrected"));
-    o.setTestEventRef(te2);
-    o.markComplete();
-    _testOrderRepo.save(o);
-    return te2;
+  public TestEvent createTestEventCorrection(TestEvent originalTestEvent) {
+    return _testEventRepo.save(
+        new TestEvent(originalTestEvent, TestCorrectionStatus.CORRECTED, "Cold feet"));
```
Comment (emmastephenson) [inline backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java:339]: 🤣

**Review by emmastephenson — APPROVED** [review]
LGTM!

**Comment by zdeveloper** [comment]
Examples from `test` env queue

Original Test Event
```json
{
    "Ordering_facility_email": "central@middle.school",
    "Ordering_facility_state": "CA",
    "Ordering_facility_street": "344 East Rocky Oak Extension",
    "Ordering_facility_street_2": "",
    "Ordering_facility_zip_code": "90210",
    "Ordering_provider_last_name": "Keith",
    "Ordering_provider_first_name": "Stewart",
    "Ordering_provider_street": "209 Cowley Avenue",
    "Ordering_provider_street_2": "",
    "Ordering_provider_city": null,
    "Ordering_provider_state": "AZ",
    "Ordering_provider_zip_code": "90210",
    "Ordering_provider_county": null,
    "Ordering_provider_phone_number": "1211849154",
    "Specimen_source_site_code": "53342003",
    "Specimen_type_code": "45345873",
    "Instrument_ID": "99f2b07f-54dc-4e9b-a762-16b479a539a9",
    "Device_ID": "Xpert Xpress SARS-CoV-2 (2)",
    "Test_date": "20220207191833",
    "Date_result_released": "20220207191833",
    "Order_test_date": "20220207191833",
    "Site_of_care": "government_agency",
    "Patient_last_name": "S",
    "Patient_first_name": "Zedd",
    "Patient_middle_name": null,
    "Patient_suffix": null,
    "Patient_race": "1002-5",
    "Patient_DOB": "19650911",
    "Patient_gender": "M",
    "Patient_ethnicity": "N",
    "Patient_street": "191 Oak Court",
    "Patient_street_2": "Eiusmod et vel optio",
    "Patient_city": "Ea vero elit molest",
    "Patient_county": "Laudantium ut atque",
    "Patient_state": "NA",
    "Patient_zip_code": "00000",
    "Patient_phone_number": "(123) 521-1234",
    "Patient_email": "zedd@skylight.digital",
    "Patient_ID": "845e6ead-63ca-481e-be53-b4e327a7decd",
    "Patient_role": "STUDENT",
    "Patient_tribal_affiliation": "",
    "Patient_preferred_language": "",
    "Employed_in_healthcare": "Y",
    "Resident_congregate_setting": "N",
    "Result_ID": "bc05560e-c1f5-416e-bac3-9bb7e0870349",
    "Corrected_result_ID": "",
    "Test_correction_reason": "",
    "Test_result_status": "F",
    "Specimen_collection_date_time": "20220207190333",
    "Ordering_provider_ID": "000",
    "First_test": "N",
    "Symptomatic_for_disease": "Y",
    "Testing_lab_name": "Central Middle School",
    "Testing_lab_CLIA": "12D3456789",
    "Testing_lab_state": "CA",
    "Testing_lab_street": "344 East Rocky Oak Extension",
    "Testing_lab_street_2": "",
    "Testing_lab_zip_code": "90210",
    "Testing_lab_county": "",
    "Testing_lab_phone_number": "(410) 956-5888",
    "Testing_lab_city": null,
    "Processing_mode_code": "P",
    "Ordering_facility_city": null,
    "Ordering_facility_county": "",
    "Ordering_facility_name": "Central Middle School",
    "Ordering_facility_phone_number": "(410) 956-5888",
    "Ordered_test_code": "94500-6",
    "Organization_name": "Dis Organization",
    "Test_result_code": "260373001",
    "Illness_onset_date": ""
}
```

Corrected Test Event
```json
{
    "Ordering_facility_email": "central@middle.school",
    "Ordering_facility_state": "CA",
    "Ordering_facility_street": "344 East Rocky Oak Extension",
    "Ordering_facility_street_2": "",
    "Ordering_facility_zip_code": "90210",
    "Ordering_provider_last_name": "Keith",
    "Ordering_provider_first_name": "Stewart",
    "Ordering_provider_street": "209 Cowley Avenue",
    "Ordering_provider_street_2": "",
    "Ordering_provider_city": null,
    "Ordering_provider_state": "AZ",
    "Ordering_provider_zip_code": "90210",
    "Ordering_provider_county": null,
    "Ordering_provider_phone_number": "1211849154",
    "Specimen_source_site_code": "53342003",
    "Specimen_type_code": "45345873",
    "Instrument_ID": "99f2b07f-54dc-4e9b-a762-16b479a539a9",
    "Device_ID": "Xpert Xpress SARS-CoV-2 (2)",
    "Test_date": "20220207192015",
    "Date_result_released": "20220207192015",
    "Order_test_date": "20220207192015",
    "Site_of_care": "government_agency",
    "Patient_last_name": "S",
    "Patient_first_name": "Zedd",
    "Patient_middle_name": null,
    "Patient_suffix": null,
    "Patient_race": "1002-5",
    "Patient_DOB": "19650911",
    "Patient_gender": "M",
    "Patient_ethnicity": "N",
    "Patient_street": "191 Oak Court",
    "Patient_street_2": "Eiusmod et vel optio",
    "Patient_city": "Ea vero elit molest",
    "Patient_county": "Laudantium ut atque",
    "Patient_state": "NA",
    "Patient_zip_code": "00000",
    "Patient_phone_number": "(123) 521-1234",
    "Patient_email": "zedd@skylight.digital",
    "Patient_ID": "845e6ead-63ca-481e-be53-b4e327a7decd",
    "Patient_role": "STUDENT",
    "Patient_tribal_affiliation": "",
    "Patient_preferred_language": "",
    "Employed_in_healthcare": "Y",
    "Resident_congregate_setting": "N",
    "Result_ID": "3b72f99d-1e25-4444-b92d-4b32f36b3033",
    "Corrected_result_ID": "bc05560e-c1f5-416e-bac3-9bb7e0870349",
    "Test_correction_reason": "This was marked as error due to data entry mistake",
    "Test_result_status": "X",
    "Specimen_collection_date_time": "20220207190515",
    "Ordering_provider_ID": "000",
    "First_test": "UNK",
    "Symptomatic_for_disease": "Y",
    "Testing_lab_name": "Central Middle School",
    "Testing_lab_CLIA": "12D3456789",
    "Testing_lab_state": "CA",
    "Testing_lab_street": "344 East Rocky Oak Extension",
    "Testing_lab_street_2": "",
    "Testing_lab_zip_code": "90210",
    "Testing_lab_county": "",
    "Testing_lab_phone_number": "(410) 956-5888",
    "Testing_lab_city": null,
    "Processing_mode_code": "P",
    "Ordering_facility_city": null,
    "Ordering_facility_county": "",
    "Ordering_facility_name": "Central Middle School",
    "Ordering_facility_phone_number": "(410) 956-5888",
    "Ordered_test_code": "94500-6",
    "Organization_name": "Dis Organization",
    "Test_result_code": "260373001",
    "Illness_onset_date": ""
}
```

**Review by nickclyde — APPROVED** [review]
Looks good! Thanks for the thorough tests.

**Review by nathancrtr — APPROVED** [review]
Looks great! Thanks for dropping an example from `test` in here, it's very helpful 🙌

**Review by alismx — APPROVED** [review]
Looks good. :test_tube:

**Pull request merged**